airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryDestinationTest.java

```diff
@@ -227,12 +231,53 @@ void testCheckSuccess(final DatasetIdResetter resetDatasetId) {
     ((ObjectNode) config).put(BigQueryConsts.CONFIG_PROJECT_ID, "fake");
     resetDatasetId.accept(config);
-    final AirbyteConnectionStatus actual = new BigQueryDestination().check(config);
-    final String actualMessage = actual.getMessage();
+
+    // Assert that check throws exception. Later it will be handled by IntegrationRunner
+    final ConfigErrorException ex = assertThrows(ConfigErrorException.class, () -> {
+      new BigQueryDestination().check(config);
+    });
+
+    final String actualMessage = ex.getMessage();
+    LOGGER.info("Checking expected failure message:" + actualMessage);
+    assertThat(actualMessage).contains("Access Denied");
+  }
+
+  @ParameterizedTest
+  @MethodSource("datasetIdResetterProvider")
+  void testCheckFailureInsufficientPermissionForCreateDataset(final DatasetIdResetter resetDatasetId) throws IOException {
+
+    if (!Files.exists(CREDENTIALS_WITHMISSED_CREATE_DATASET_ROLE_PATH)) {
+      throw new IllegalStateException("""
+                                      Json config not found. Must provide path to a big query credentials file,
+                                       please add file with creds to
+                                      ../destination-bigquery/secrets/credentialsWithMissedDatasetCreationRole.json.""");
+    }
+    final String fullConfigAsString = Files.readString(CREDENTIALS_WITHMISSED_CREATE_DATASET_ROLE_PATH);
+    final JsonNode credentialsJson = Jsons.deserialize(fullConfigAsString).get(BigQueryConsts.BIGQUERY_BASIC_CONFIG);
+    final String projectId = credentialsJson.get(BigQueryConsts.CONFIG_PROJECT_ID).asText();
+    final String datasetId = Strings.addRandomSuffix(DATASET_NAME_PREFIX, "_", 8);
+
+    final JsonNode insufficientRoleConfig;
+
+    insufficientRoleConfig = Jsons.jsonNode(ImmutableMap.builder()
+        .put(BigQueryConsts.CONFIG_PROJECT_ID, projectId)
+        .put(BigQueryConsts.CONFIG_CREDS, credentialsJson.toString())
+        .put(BigQueryConsts.CONFIG_DATASET_ID, datasetId)
+        .put(BigQueryConsts.CONFIG_DATASET_LOCATION, DATASET_LOCATION)
+        .put(BIG_QUERY_CLIENT_CHUNK_SIZE, 10)
+        .build());
+
+    resetDatasetId.accept(insufficientRoleConfig);
+
+    // Assert that check throws exception. Later it will be handled by IntegrationRunner
+    final ConfigErrorException ex = assertThrows(ConfigErrorException.class, () -> {
+      new BigQueryDestination().check(insufficientRoleConfig);
+    });
+
+    final String actualMessage = ex.getMessage();
     LOGGER.info("Checking expected failure message:" + actualMessage);
-    assertTrue(actualMessage.contains("Access Denied:"));
-    final AirbyteConnectionStatus expected = new AirbyteConnectionStatus().withStatus(Status.FAILED).withMessage("");
-    assertEquals(expected, actual.withMessage(""));
+    assertThat(actualMessage)
+        .contains("Project dataline-integration-testing: User does not have bigquery.datasets.create permission");
```

Comment:
I would probably remove the `dataline-integration-testing:` portion since that's really internal to Airbyte and better to keep this not directly exposed since it provides no value to anybody else

Also same comment as above, if the `LOGGER` class is not needed then this `assertThat` can be simplified into one line

Reply:
updated, thanks